src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt

```diff
@@ -253,6 +253,7 @@ private constructor(
 
           ChatMessage(speaker = parsed, message.text)
         }
+    this.messages.addAll(chatMessages)
```

Comment:
This is needed and reasonable too. Let's add all these message in advance. `addMessageAtIndex` will not even execute in `receiveMessage` as we check the condition: `sessionMessage != incomingMessage`.